docs(website): update pattern contribution guidelines (#1126)
twilio-labs/paste: packages/paste-website/src/pages/patterns/contribute/index.mdx

<content>

Thanks for your interest in contributing patterns! Since this is a general guideline, feel free to grow or shrink this process however you need.

> ## General process:

Reviewer: ```suggestion
## General process
```

Have feedback on the process? Add to [our feedback doc](https://docs.google.com/document/d/1N_edYt6O9_2DpJi0kwvU0Fjaxur-voEOi89YzP2qyOE/edit#).

## Roles

### Design System’s role
We’re here to partner with you throughout the pattern process to ensure you’re building something that is systemized and accessible. We’ll also help you add to the docs site, refine design assets, and create coded examples. 

### Your role (anyone at Twilio!)
You’ll create and research the doc/pattern/artifact and socialize it within your crit pods, committees, and peers. PDs have great insight into product-specific customer problems, which means if you’re a PD, you’re especially primed to contribute a new pattern or update an existing one.

## Contribution types

Patterns are for you and your teammates, so it’s up to you what scope and type of contribution will have the most impact.

### Ways you can contribute

  - Creating a full, net-new pattern
  
  - Creating a variation of a pattern
  
  - Documenting one part of a pattern
  
  - Sharing research related to a pattern
  
  - Refining an existing pattern based on new information

### Where do I start?

> Check out our [list of most requested patterns](https://paste.twilio.design/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and the `#help-design-system` channel to see if there’s a need for something else.

Reviewer: ```suggestion
Check out our [list of most requested patterns](/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and the `#help-design-system` channel to see if there’s a need for something else.
```

> Check out our [list of most requested patterns](https://paste.twilio.design/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and the `#help-design-system` channel to see if there’s a need for something else.

Reviewer: ```suggestion
Check out our [list of most requested patterns](https://paste.twilio.design/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and create a [Github Discussion](https://github.com/twilio-labs/paste/discussions) to see if there’s a need for something else.
```

> Check out our [list of most requested patterns](https://paste.twilio.design/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and create a [Github Discussion](https://github.com/twilio-labs/paste/discussions) to see if there’s a need for something else.

Reviewer: ```suggestion
Check out our [list of most requested patterns](/patterns#patterns) to see what's available. <strong>Have a new idea?</strong> Reach out to your peers and create a [Github Discussion](https://github.com/twilio-labs/paste/discussions) to see if there’s a need for something else.
```
